timdex_dataset_api/metadata.py

```diff
@@ -183,3 +274,39 @@ def _attach_database_file(self, conn: DuckDBPyConnection) -> None:
         conn.execute(
             f"""attach '{self.metadata_database_path}' AS static_db (READ_ONLY);"""
         )
+
+    def write_append_delta_duckdb(self, filepath: str) -> None:
+        """Write an append delta for an ETL parquet file.
+
+        A DuckDB context is used to both read metadata-only columns from the ETL parquet
+        file, then write an append delta parquet file to /metadata/append_deltas.  The
+        write is performed by DuckDB's COPY function.
+
+        Note: this operation is safe in parallel with other possible append delta writes.
+        """
+        start_time = time.perf_counter()
+
+        output_path = f"{self.append_deltas_path}/append_delta-{filepath.split('/')[-1]}"
+
+        # ensure s3:// schema prefix is present
+        if self.location_scheme == "s3":
+            filepath = f"s3://{filepath.removeprefix("s3://")}"
+
+        # perform query + write as one SQL statement
+        sql = f"""
+        copy (
+            select
+                {','.join(ORDERED_METADATA_COLUMN_NAMES)}
+            from read_parquet(
+                '{filepath}',
+                hive_partitioning=true,
+                filename=true
+            )
+        ) to '{output_path}'
+        (FORMAT parquet);
```

Comment:
This is dense but important DuckDB SQL here.  

The `copy` command takes the results from the query in parenthesis, and sends those results to the output (in our case a parquet file).  I wish they had called it `pipe()`...